From 2a0a8030643b56659232931f4a41beb2af268d53 Mon Sep 17 00:00:00 2001
From: Hari Chandrakanthan <quic_haric@quicinc.com>
Date: Fri, 21 Jan 2022 19:12:23 +0530
Subject: [PATCH] sigma-dut: increase twt sp guard interval

Increase the twt sp guard interval to 1000us to accomodate
more ppdus within the the twt service period.

Without this change, less number of ppdus are observed within the
twt service period.

Signed-off-by: Hari Chandrakanthan <quic_haric@quicinc.com>
---
 ap.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ap.c b/ap.c
index 3fe8639..f629d55 100644
--- a/ap.c
+++ b/ap.c
@@ -7992,6 +7992,9 @@ static void fwtest_set_he_params(struct sigma_dut *dut, const char *ifname)
 		fwtest_cmd_wrapper(dut, "-m 0x48 -v 0 63 1000", ifname);
 	}
 
+	/* sp_guard_interval_in_us WAR */
+	fwtest_cmd_wrapper(dut, "-m 0x47 -v 0 398 1000", ifname);
+
 	if (dut->ap_he_ppdu == PPDU_MU &&
 	    dut->ap_he_dlofdma == VALUE_ENABLED) {
 		fwtest_cmd_wrapper(dut, "-m 0x00 -v 0 32770 217", ifname);
-- 
2.7.4

